### ISSUE TITLE
Tinning bug: "You make, but cannot pick up, a bag"

### ISSUE THREAD

**Issue by paulcee** [NetHack/NetHack#1021]
NetHack 3.6.6 (tiles) on Windows 10: I was in the Gnomish Mines carrying (among other things) a tinning kit, and a bag called GOOD. I stepped on a hobbit corpse and tinned it. This evident bug occurred:

There is a hobbit corpse here; tin it? [ynq]
You make, but cannot pick up, a bag called GOOD.

I checked the contents of the square and they were a sling and a tin of hobbit meat. No bag -- that was still in my inventory. So this seems like a string pointer bug.

**Comment by pat-rankin** [comment]
Do you have perm_invent enabled?  Whenever the persistent inventory window gets updated, every item in inventory gets reformatted and that churns through a set of temporary buffers used to hold the formatted text.  If something was formatted before the inventory update but not displayed until after, it can end up with the wrong text (from some other item that reused its buffer).

This was fixed for to-be-3.7 nearly two years ago (commit e43ec0cef1cc7c4e639c7de7e1d9ff26c6c8c629); I don't think that the fixed ever made it into 3.6.x.

**Comment by paulcee** [comment]
Yes, perm_invent is enabled; so it sounds like a fix is ready for the future. Thanks.
